components/scream/src/interface/scream_scorpio_interface.F90

```diff
@@ -0,0 +1,977 @@
+      curr => prev%next
+    end do
+    allocate(prev%next)
+    curr => prev%next
+    allocate(curr%coord)
+    hist_coord => curr%coord
+    pio_atm_file%dimcounter = pio_atm_file%dimcounter + 1
+    ! Register this dimension
+    hist_coord%name      = trim(shortname)
+    hist_coord%long_name = trim(longname)
+    hist_coord%dimsize   = length
+    if (length.eq.0) then
+      ierr = PIO_def_dim(pio_atm_file%pioFileDesc, trim(shortname), pio_unlimited , hist_coord%dimid)
+    else
+      ierr = PIO_def_dim(pio_atm_file%pioFileDesc, trim(shortname), length , hist_coord%dimid)
+    end if
+    call errorHandle("PIO ERROR: could not define dimension "//trim(shortname)//" on file: "//trim(pio_atm_filename),ierr)
+    
+    return
+  end subroutine register_dimension
+!=====================================================================!
+  ! Register a variable with a specific pio output file
+  subroutine register_variable(pio_atm_filename,shortname,longname,numdims,var_dimensions,dtype,pio_decomp_tag)
+    character(len=*), intent(in) :: pio_atm_filename         ! Name of the file to register this variable with
+    character(len=*), intent(in) :: shortname,longname       ! short and long names for the variable.  Short: variable name in file, Long: more descriptive name
+    integer, intent(in)          :: numdims                  ! Number of dimensions for this variable, including time dimension
+    character(len=*), intent(in) :: var_dimensions(numdims)  ! String array with shortname descriptors for each dimension of variable.
+    integer, intent(in)          :: dtype                    ! datatype for this variable, REAL, DOUBLE, INTEGER, etc.
+    character(len=*), intent(in) :: pio_decomp_tag           ! Unique tag for this variables decomposition type, to be used to determine if the io-decomp already exists.
+
+    ! Local variables
+    type(pio_atm_output),pointer :: pio_atm_file
+    integer                      :: loc_len
+    logical                      :: found
+    type(io_desc_t), pointer     :: iodesc_loc
+    type(hist_var_t), pointer    :: hist_var
+    integer                      :: dim_ii
+    integer                      :: ierr
+    integer, allocatable         :: dimlen(:)
+    integer                      :: total_dimlen, my_dof_len, extra_procs
+    integer                      :: ii, istart, istop
+    logical                      :: has_t_dim  ! Logical to flag whether this variable has a time-dimension.  This is important for the decomposition step.
+
+    type(hist_var_list), pointer :: curr => null(), prev => null()
+ 
+    ! Find the pointer for this file
+    call get_pio_atm_File(trim(pio_atm_filename),pio_atm_file)
+    ! Update the number of variables on file 
+    pio_atm_file%varcounter = pio_atm_file%varcounter + 1
+
+    ! Get a new variable pointer in var_list
+    curr => pio_atm_file%var_list_top
+    do while (associated(curr))
+      prev => curr
+      curr => prev%next
+    end do
```

Comment:
would it make sense to check that the new variable isn't already defined since you're looping over all the variable names already?

Reply:
Yes, that's a great idea